codegen/src/main/java/software/amazon/awssdk/codegen/model/service/ServiceMetadata.java

```diff
@@ -111,6 +113,14 @@ public void setProtocol(String protocol) {
         this.protocol = protocol;
     }
 
+    public List<String> getProtocols() {
```

Comment:
Can we deprecate `getProtocol()` method and add documentation to link to ProtocolUtils.resolveProtocol?

Should we add checkstyle rule to prevent the use of getProtocol so that we don't use it unintentionally in the future?

Reply:
Updated to deprecate getProtocol() , and added cbor check for Kinesis customization

There are other getProtocol() methods of same name in other classes so can't use simple checkstyle rule, would need to write custom class. I think the deprecation should be fine for now, wdyt?